tracing-attributes/src/lib.rs

```diff
@@ -363,13 +419,19 @@ fn gen_body(
             // }
             // ```
             .map(|x| {
-                // if we are inside a function generated by async-trait, we
-                // should take care to rewrite "_self" as "self" for
-                // 'user convenience'
-                if async_trait_fun.is_some() && x == "_self" {
+                // if we are inside a function generated by async-trait <=0.1.43, we need to
+                // take care to rewrite "_self" as "self" for 'user convenience'
+                if self_type.is_some() && x == "_self" {
                     (Ident::new("self", x.span()), x)
                 } else {
-                    (x.clone(), x)
+                    match variables_rebindings.iter().find(|(_, old)| old == &x) {
+                        Some((new, old)) => {
+                            // this inversion of 'old' and 'new' is deliberate, in order to
+                            // compensate for async-trait renaming variables
```

Comment:
tiny nitpick: it could be nice to explain _how_ `async-trait` renames variables that we have to compensate for? not a blocker at all!

Reply:
This was actually a remnant of the code that was added when I was trying out thze `_assync_trait` marker, I deleted that code, which simplified a bit functions signatures in the process ;)